File: C/spiral_matrix.c
#include<stdio.h>

void printSpiral(int rows, int cols, int matrix[rows][cols]) {
    int top = 0, left = 0;
    int bottom = rows - 1, right = cols - 1;
    
    printf("\nSpiral Order:\n");
    while (top <= bottom && left <= right) {
        for(int i = left; i <= right; i++){
            printf("%d ", matrix[top][i]);
        }
        top++;

        for(int i = top; i <= bottom; i++){
            printf("%d ", matrix[i][right]);
        }
        right--;

        if(top <= bottom) {
            // Print bottom row
            for(int i = right; i >= left; i--){
                printf("%d ", matrix[bottom][i]);
            }
            bottom--;
        }

        if(left <= right) {
            // Print left column
            for(int i = bottom; i >= top; i--){
                printf("%d ", matrix[i][left]);
            }
            left++;
        }
    }
}

int main() {
    int rows, cols;
    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int matrix[rows][cols];
    
    printf("Enter the matrix elements:\n");
    for(int i = 0; i < rows; i++) {
        for(int j = 0; j < cols; j++) {
            scanf("%d", &matrix[i][j]);
        }
    }
    
    printf("\nOriginal Matrix:\n");
    for(int i = 0; i < rows; i++) {
        for(int j = 0; j < cols; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }

    printSpiral(rows, cols, matrix);
    return 0;
}

File: Python/spiral_matrix.py
def spiral_order(matrix):
    result = []
    top, bottom = 0, len(matrix)
    left, right = 0, len(matrix[0])

    while top < bottom and left < right:
        for i in range(left, right):
            result.append(matrix[top][i])
        top += 1

        for i in range(top, bottom):
            result.append(matrix[i][right - 1])
        right -= 1

        if top < bottom:
            for i in range(right - 1, left - 1, -1):
                result.append(matrix[bottom - 1][i])
            bottom -= 1

        if left < right:
            for i in range(bottom - 1, top - 1, -1):
                result.append(matrix[i][left])
            left += 1

    return result

# Input from user
r = int(input("Enter number of rows: "))
c = int(input("Enter number of columns: "))
matrix = []

print("Enter matrix row by row:")
for i in range(r):
    row = list(map(int, input().split()))
    matrix.append(row)

print("Spiral order:", spiral_order(matrix))
